src/error.rs

```diff
@@ -0,0 +1,75 @@
+pub use app::RuntimeError;
+pub use css::{CssParseError, DynamicCssParseError, HotReloadError};
+pub use css_parser::{
+    CssBackgroundParseError, CssBorderParseError, CssBorderRadiusParseError, CssColorParseError,
+    CssDirectionParseError, CssFontFamilyParseError, CssGradientStopParseError, CssImageParseError,
+    CssMetric, CssParsingError, CssShadowParseError, CssShapeParseError, InvalidValueErr,
+    PercentageParseError, PixelParseError,
+};
+pub use font::FontError;
+pub use image::ImageError;
+pub use simplecss::Error as CssSyntaxError;
+
+// TODO: re-export the sub-types of ClipboardError!
+pub use clipboard2::ClipboardError;
+
+pub use widgets::errors::*;
+pub use window::WindowCreateError;
+
+#[derive(Debug)]
+pub enum Error<'a> {
+    CssParse(CssParseError<'a>),
+    Font(FontError),
+    Image(ImageError),
+    Clipboard(ClipboardError),
+    WindowCreate(WindowCreateError),
+    HotReload(HotReloadError)
+}
+
+impl<'a> From<CssParseError<'a>> for Error<'a> {
+    fn from(e: CssParseError<'a>) -> Error {
+        Error::CssParse(e)
+    }
+}
+
+impl<'a> From<FontError> for Error<'a> {
+    fn from(e: FontError) -> Error<'a> {
+        Error::Font(e)
+    }
+}
+
+impl<'a> From<ImageError> for Error<'a> {
+    fn from(e: ImageError) -> Error<'a> {
+        Error::Image(e)
+    }
+}
+
+impl<'a> From<ClipboardError> for Error<'a> {
```

Comment:
Is it possible to use `impl_from!` - I mean we already use it everwhere else?

Reply:
Not really. Another macro would have to be created.

`impl_from!` only works
- From a type with a lifetime to a type which also has a lifetime
- From a type without a lifetime to a type which also does not have a lifetime

ClipboardError does not have a lifetime but Error does.

(This was already the case with your two separate `macro_rules!`, the only thing I did was combine them)

Reply:
Alright, not a big deal, then just leave it that way.